### ISSUE TITLE
.eslintrc is incorrectly resolved from within another .eslintrc file

### ISSUE THREAD

**Issue by remcohaszing** [eslint/eslint#6479]
**What version of ESLint are you using?**
2.13.0

**What parser (default, Babel-ESLint, etc.) are you using?**
None

**Please show your full configuration:**
The following directory tree:
```
|- src/
|   `- .eslintrc
`-  website/
     ` app/
        ` .eslintrc
```

**What did you do? Please include the actual source code causing the issue.**
*website/app/.eslintrc* contains the following:
```yaml
extends: ../../src/.eslintrc
```

**What did you expect to happen?**
*website/app/.eslintrc* to extend *src/.eslintrc*. This does work when using `eslint@2.12.0`.

A workaround is to incorrectly remove one `../` from the path.

**What actually happened? Please include the actual, raw output from ESLint.**
```
/home/remco/d-centralize/appsemble/node_modules/eslint/lib/config/config-file.js:402
            throw e;
            ^

Error: Cannot read config file: /home/remco/d-centralize/src/.eslintrc
Error: ENOENT: no such file or directory, open '/home/remco/d-centralize/src/.eslintrc'
Referenced from: /home/remco/d-centralize/appsemble/website/app/.eslintrc
    at Error (native)
    at Object.fs.openSync (fs.js:634:18)
    at Object.fs.readFileSync (fs.js:502:33)
    at readFile (/home/remco/d-centralize/appsemble/node_modules/eslint/lib/config/config-file.js:71:15)
    at loadLegacyConfigFile (/home/remco/d-centralize/appsemble/node_modules/eslint/lib/config/config-file.js:143:44)
    at loadConfigFile (/home/remco/d-centralize/appsemble/node_modules/eslint/lib/config/config-file.js:223:22)
    at load (/home/remco/d-centralize/appsemble/node_modules/eslint/lib/config/config-file.js:500:18)
    at /home/remco/d-centralize/appsemble/node_modules/eslint/lib/config/config-file.js:392:36
    at Array.reduceRight (native)
    at applyExtends (/home/remco/d-centralize/appsemble/node_modules/eslint/lib/config/config-file.js:363:28)
    at Object.load (/home/remco/d-centralize/appsemble/node_modules/eslint/lib/config/config-file.js:531:22)
    at loadConfig (/home/remco/d-centralize/appsemble/node_modules/eslint/lib/config.js:64:33)
    at getLocalConfig (/home/remco/d-centralize/appsemble/node_modules/eslint/lib/config.js:126:23)
    at Config.getConfig (/home/remco/d-centralize/appsemble/node_modules/eslint/lib/config.js:227:22)
    at processText (/home/remco/d-centralize/appsemble/node_modules/eslint/lib/cli-engine.js:223:27)
    at CLIEngine.executeOnText (/home/remco/d-centralize/appsemble/node_modules/eslint/lib/cli-engine.js:761:26)
```

**Comment by SeanSobey** [comment]
Getting this same issue, seems related to the updates in config-file.js method applyExtends(config, filePath, relativeTo);. Specifically line 384: 
` parentPath = (!isAbsolutePath(parentPath) ?
                path.join(relativeTo || path.dirname(filePath), parentPath) :
                parentPath
            );`

**Comment by kokarn** [comment]
This breaks with this as well with the same error.
```
"extends": "./default.eslintrc",
```

**Comment by kokarn** [comment]
Confirmed it's this commit that breaks https://github.com/eslint/eslint/commit/fb49c7fb119c6d2a2a0441a01190bda8a9172353 from pull https://github.com/eslint/eslint/pull/6359

**Comment by kokarn** [comment]
Also, this is a duplicate of #6482 so this can be closed ^^

**Comment by vitorbal** [comment]
Closing this as it is a duplicate of #6450. There should be a patch release today with the fix. Follow #6450 for updates!
